### PR TITLE
fix: use map to verify unordered events in the scheduing framework

### LINKED ISSUE #80310: Scheduler's bind extension point is flaky
**What happened**:
Bind extension point in the scheduler framework is flaky

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/80305/pull-kubernetes-integration/1151863322098798593/

/sig scheduling
/kind flake

### PR DESCRIPTION
**Pull request by draveness**
/kind bug
/priority important-soon
/assign @bsalamat @ahg-g @chenchun 
/sig scheduling

**What this PR does / why we need it**:

Use a map to verify unordered events in the scheduling framework.

Currently, the integration tests verify the events with the order which predefined in the slice, but due to concurrency issue, the events could come with a different order.

```go
framework_test.go:954: test #3: Expect invoke event 1 from plugin mock-unreserve-plugin instead of bind-plugin-1
framework_test.go:957: test #3: Expect val of invoke event 1 to be 1 instead of 2
framework_test.go:954: test #3: Expect invoke event 2 from plugin bind-plugin-1 instead of mock-unreserve-plugin
framework_test.go:957: test #3: Expect val of invoke event 2 to be 2 instead of 1
```

**Which issue(s) this PR fixes**:

Fixes #80310

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by ahg-g** [comment]
do we still need this given #80341?

File: test/integration/scheduler/framework_test.go [hunk 1]
```diff
@@ -944,18 +948,32 @@ func TestBindPlugin(t *testing.T) {
 				t.Errorf("test #%v: Didn't expected the postbind plugin to be called %d times.", i, postbindPlugin.numPostbindCalled)
 			}
 		}
+
+		expectInvokeEvents := map[pluginInvokeEvent]struct{}{}
```
Comment (bsalamat) [inline test/integration/scheduler/framework_test.go]: The invocation of bind and unreserve plugins must happen in a specific order. The fact that we see them happening in different orders could possibly point to a bug in the implementation.

Comment (draveness) [reply test/integration/scheduler/framework_test.go]: Let's see if it still happens after #80341 merged into master.

**Comment by draveness** [comment]
/close

ref: #80341

**Pull request closed without merging**